charts/workload/templates/deployment.yaml

```
@@ -2,22 +2,30 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ include "workload.name" . }}
+  {{- if and .Values.workload .Values.workload.annotations }}
   annotations:
     {{- toYaml .Values.workload.annotations | nindent 4 }}
+  {{- end }}
   labels:
     {{- include "workload.commonLabels" . | nindent 4 }}
+    {{- if and .Values.workload .Values.workload.labels }}
     {{- toYaml .Values.workload.labels | nindent 4 }}
+    {{- end }}
 spec:
   selector:
     matchLabels:
       {{- include "workload.selectorLabels" . | nindent 6 }}
   template:
     metadata:
+      {{- if and .Values.workload .Values.workload.containers .Values.workload.containers.annotations }}
```

nitpick: you can also do this with dig I think (here '{}') is the default 
```suggestion
      {{- if dig "workload" "containers" "annotations" {} .Values }}
```

---

Didn't know about that one, good idea, but for future PR in order to fix the current issue. Thanks!